scipy/optimize/direct.c

```diff
@@ -0,0 +1,86 @@
+#include "Python.h"
+#include "numpy/arrayobject.h"
+#include "direct.h"
+#define MY_ALLOC(p, t, n, ret_code) p = (t *) malloc(sizeof(t) * (n)); \
+                          if (!(p)) { *ret_code = DIRECT_OUT_OF_MEMORY; }
+#define MY_FREE(p) if (p) free(p)
+
+static PyObject *
+direct(PyObject *self, PyObject *args)
+{
+    PyObject *f, *f_args, *lb, *ub, *callback;
+    int dimension, max_feval, max_iter, force_stop, disp;
+    const double *lower_bounds, *upper_bounds;
+    double minf, magic_eps, magic_eps_abs, *x;
+    double volume_reltol, sigma_reltol;
+    double fglobal, fglobal_reltol;
+    FILE *logfile = NULL;
+    direct_algorithm algorithm;
+    direct_return_code ret_code;
+
+    if (!PyArg_ParseTuple(args, "OOOOidiiiddddO",
+                          &f, &lb, &ub, &f_args, &disp, &magic_eps,
+                          &max_feval, &max_iter, (int*) &algorithm,
+                          &fglobal, &fglobal_reltol,
+                          &volume_reltol, &sigma_reltol, &callback))
+    {
+        return NULL;
+    }
+
+    if (disp) {
```

Comment:
I thought the review comments said that `disp` was disabled, because it changed outputs?. So can this code either be removed, or at least a comment about it added?

Reply:
As far as I understand the @dschmitz89 's intent in https://github.com/scipy/scipy/pull/14300#issuecomment-1086929270, I think he wanted to remove `disp` from the API or in other words not expose this functionality as of now. However, we may keep the implementation which uses `disp`, so that in future we don't need to re-implement this functionality from scratch. Is this the correct interpretation of your comment, @dschmitz89 ?

Reply:
yes exactly. `disp` is set to `False` in the Python interface currently. But the functionality could be debugged in the C implementation in future potentially, so I suggest just adding a comment also in the C code.

Reply:
Okay, sounds good - let's add a comment.

Reply:
Okay. I will add a comment in the code.